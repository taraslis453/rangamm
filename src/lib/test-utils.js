import React from 'react'
import {render} from '@testing-library/react'
import {theme} from '../theme'
import {ThemeProvider} from 'styled-components'
import {BrowserRouter} from 'react-router-dom'
import {store} from '../store'
import {Provider} from 'react-redux'
const AllTheProviders = ({children}) => {
  return (
    <div id='root'>
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </Provider>
      </BrowserRouter>
    </div>
  )
}

const customRender = (ui, options) => render(ui, {wrapper: AllTheProviders, ...options})
export * from '@testing-library/react'

export {customRender as render}
