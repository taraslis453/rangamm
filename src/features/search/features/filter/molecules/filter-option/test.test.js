import React from 'react'
import {render} from 'lib/test-utils'
import {FilterOption} from './index'
describe('<FilterOption/>', () => {
  it('renders properly', () => {
    const {container} = render(
      <>
        <FilterOption
          name={'order_by'}
          id={`radio-0`}
          label='Relevant'
          value='relevant'
          onChange={() => console.log('onChange')}
          checked={true}
        />
        <FilterOption
          name={'order_by'}
          id={`radio-1`}
          label='Latest'
          value='latest'
          onChange={() => console.log('onChange')}
          checked={false}
        />
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
