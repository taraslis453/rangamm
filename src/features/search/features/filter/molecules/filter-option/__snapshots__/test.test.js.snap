// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterOption/> renders properly 1`] = `
<div
  id="root"
>
  <label
    class="sc-hKgILt meBvN"
    cursor="pointer"
    display="flex"
    for="radio-0"
  >
    <input
      class="sc-fubCfw ktfAJq"
      id="radio-0"
      name="order_by"
      type="radio"
      value="relevant"
    />
    <div
      class="sc-iBPRYJ foHfmu"
      data-testid="filterIconChecked"
    >
      <div>
        <div>
          <div
            data-src="tick.svg"
            src=""
          />
        </div>
      </div>
    </div>
    <div
      class="sc-gKsewC iOtxnD"
      color="black"
      font-size="1"
    >
      Relevant
    </div>
  </label>
  <label
    class="sc-hKgILt meBvN"
    cursor="pointer"
    display="flex"
    for="radio-1"
  >
    <input
      class="sc-fubCfw ktfAJq"
      id="radio-1"
      name="order_by"
      type="radio"
      value="latest"
    />
    <div
      class="sc-gKsewC lpsDW"
      color="#767676"
      font-size="1"
    >
      Latest
    </div>
  </label>
</div>
`;
